core/src/test/scala/org/apache/spark/util/UtilsSuite.scala

```diff
@@ -731,6 +732,52 @@ class UtilsSuite extends SparkFunSuite with ResetSystemProperties {
     assert(!sourceFile2.exists())
   }
 
+  test("SPARK-50716: deleteRecursively - SymbolicLink To File") {
```

Comment:
The addition of this test is solely to improve coverage. 

Without the current pr, since the `BasicFileAttributes` of a file are read without using `LinkOption.NOFOLLOW_LINKS`, for symbolic links, the `BasicFileAttributes` of the corresponding file are read instead. 

As a result, it is treated as a RegularFile and deleted.